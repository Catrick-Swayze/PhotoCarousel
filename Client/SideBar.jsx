import React from 'react';
import SideBarEntry from './sideBarEntry.jsx';

const SideBar = (props) => (
<div id="sideBar">
   {props.imageList.map(image =>
      <SideBarEntry
        image={image}
      />
    )}
</div>

)

export default SideBar;