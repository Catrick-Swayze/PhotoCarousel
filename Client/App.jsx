import React from 'react';
import MainImage from './mainImage.jsx'
import SideBar from './sideBar.jsx';

class App extends React.Component {
  constructor(props) {
    super(props)
      this.state = {
        imageList: []
      }
  }




  getImagesForEndpoint() {
    fetch('http://localhost:8080/api/products/standard-fit-hoodied-sweatshirt/baby-blue/carousel')
    .then(res => res.json())
    .then((images) => {

      this.setState({
        imageList: images
      })
    })
    .catch(err => console.log(err))
  }


  componentDidMount() {
    this.getImagesForEndpoint()
  }

  render () {

    return (
      <div>
        <div>
          <SideBar imageList={this.state.imageList}/>
        </div>
        <MainImage />

      </div>
    )

  }
}

export default App;