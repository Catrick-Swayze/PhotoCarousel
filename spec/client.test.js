import 'jsdom-global/register';
import React from 'react';
import { shallow } from 'enzyme';
import App from '../Client/App.jsx';
import SideBar from '../Client/SideBar.jsx';
import SideBarEntry from '../Client/sideBarEntry.jsx';
import { mount } from 'enzyme';
import MainImage from '../Client/mainImage.jsx'
const faker = require('faker');



test('truth serum', () => {
  expect('true').toBe('true');
});

var mockImageDataCreator = function(numberOfEnries) {

    var dataEntries = [];
    for (var i = 0; i < numberOfEnries; i++) {
      dataEntries.push({
         product: faker.commerce.productName().split(' ').join('-'),
         imageName: faker.commerce.productDescription(),
         color: faker.commerce.color().split(' ').join('-'),
         url: 'https://target-image-carousel.s3-us-west-1.amazonaws.com/Screen+Shot+2020-09-24+at+5.21.25+PM.png',
         alt: faker.image.abstract()
      })
    }
    return dataEntries;
  }

  var mockImages = mockImageDataCreator(7);
  var mockSideBarImages = mockImages.slice(0, 5)
  var mockOneImage = mockImages[0];


describe('App', () => {

  test('should render the App component correctly', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot();
  })
  test('should render SideBar', () => {
    const wrapper = shallow(<SideBar imageList={mockSideBarImages}/>);
    expect(wrapper.exists()).toBe(true);
  })

  test('should render one main image', () => {
    const wrapper = mount(<MainImage image={mockOneImage}/>);
    expect(wrapper.find('MainImage')).toHaveLength(1);
  })
})


describe('SideBar', () => {
  test('should render 5 sidebar entry image components', () => {
    const wrapper = mount(<SideBar imageList={mockSideBarImages}/>);
    expect(wrapper.find('SideBarEntry')).toHaveLength(5);
  })
  test('each sidebar entry should have an img tag', () => {
    const wrapper = mount (<SideBarEntry image={mockOneImage}/>);
   expect(wrapper.find('img')).toHaveLength(1)
  })

})
