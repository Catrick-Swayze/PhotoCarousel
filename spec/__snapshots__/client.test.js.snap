// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component correctly 1`] = `<div />`;

exports[`PopOut should render the PopOut component correctly 1`] = `
<div>
  <div
    className="bigCarousel"
  >
    <LastPicture />
    <BigImage />
    <div>
      <Exit />
      <NextPicture />
    </div>
  </div>
  <div>
    <BottomBar />
  </div>
</div>
`;
